surefire-provider/src/main/java/org/arquillian/smart/testing/surefire/provider/SmartTestingSurefireProvider.java

```diff
@@ -45,10 +46,15 @@ private TestsToRun getOptimizedTestsToRun(TestsToRun testsToRun) {
                 return testsToRun.getClassByName(className) != null;
             });
 
-        return new TestStrategyApplier(testsToRun, testExecutionPlannerLoader, bootParams.getTestClassLoader()).apply(
+        return new TestStrategyApplier(testsToRun, testExecutionPlannerLoader, bootParams.getTestClassLoader(), getBasedir()).apply(
```

Comment:
We need to inverse the control of creating those object. Things we are explicitly passing around is getting too big. I will work on the DI.